booking.js

```diff
@@ -84,23 +91,45 @@ bookingButtonOne.addEventListener("click", async () => {
     document
         .querySelector(".booking-container__submit-button")
         .classList.add("fade-in");
-    //Sets variable value to the userinput of .booking-container__dateInput
-    let dateInput = document.querySelector(
-        ".booking-container__dateInput"
-    ).value;
+
     /*Adds variables into url and runs function fetchData then 
     puts return value into dateApi variable*/
     dateApi = await fetchData(
-        `https://lernia-sjj-assignments.vercel.app/api/booking/available-times?date=${dateInput}&challenge=${cardId}`
+        `https://lernia-sjj-assignments.vercel.app/api/booking/available-times?date=${dateValue}&challenge=${cardId}`
     );
 
     //giving the date key the correct value.
-    booking.date = dateInput;
+    booking.date = dateValue;
 
     createAvailableTimes(dateApi);
 });
 
 submitBookingButton.addEventListener("click", async () => {
+    // getting the values from the different inputs into the object.
+    booking.name = nameInput.value;
+    booking.email = emailInput.value;
+    booking.participants = parseInt(particiSelect.value);
+    booking.time = timeInput.value;
+    //Function that checks that the email has the right format before submitting
+    function validateEmail(email) {
+        const atPos = email.indexOf("@");
+        const dotPos = email.lastIndexOf(".");
+        return atPos > 0 && dotPos > atPos + 1 && dotPos < email.length - 1;
+    }
```

Comment:
This is a nice fuction for validating email. It's not complete, but good enough to catch most misspellings etc.